README.md

```diff
@@ -10,12 +10,328 @@ A core widget library for Dojo 2.
+    	- [d](#d)
+    	- [Widgets with Children](#widgets-with-children)
+    - [Extending Base Widget](#extending-base-widget)
+    - [Projector](#projector)
+    - [Dojo Widget Components](#dojo-widget-components)
+- [How do I contribute?](#how-do-i-contribute)
+    - [Installation](#installation)
+    - [Testing](#testing)
+- [Licensing information](#licensing-information)
+
+## Usage
+
+To use dojo-widgets in a project install the package along with the required peer dependencies.
+
+```shell
+npm install dojo-widgets
+
+# peer dependencies
+npm install dojo-has
+npm install dojo-shim
+npm install dojo-core
+npm install dojo-compose
+npm install dojo-store
+```
+
+To use `dojo-widgets` import the module in the project. For more details see [features](#features) below.
+
+```ts
+import createButton from 'dojo-widgets/components/createButton';
+```
+
+or use the [dojo cli](https://github.com/dojo/cli-create-app) to create a complete Dojo 2 skeleton application.
+
 ## Features
 
 Dojo 2 Widgets are based on a virtual DOM implementation named [Maquette](http://maquettejs.org/) as well as some foundational classes
 provided in [dojo-compose](https://github.com/dojo/compose).
 
-To see how the widgets are utilised, there is an [example application](https://github.com/kitsonk/app-example) available.  Currently, outside of the source distribution itself, there is not a lot of documentation, as the code base is under heavy development.
+The examples below are provided in TypeScript syntax. The package does work under JavaScript, but for clarity, the examples will only include one syntax.
+
+### Base Widget
+
+The class `createWidgetBase` provides all base Dojo 2 widget functionality including caching and widget lifecycle management. It can be used directly or extended to create custom widgets.
+
+To customise the widget an optional `options` argument can be provided with the following interface.
+
+**Type**: `WidgetOptions<WidgetState>` - All properties are optional.
+
+|Property|Type|Description|
+|---|---|---|
+|id|string|identifier for the widget|
+|state|WidgetState|Initial state of the widget|
+|stateFrom|StoreObservablePatchable|Observable that provides state for the widget|
+|listeners|EventedListenersMap|Map of listeners for to attach to the widget|
+|tagName|string|Override the widgets     tagname|
+|getChildrenNodes|Function|Function that returns an array of children DNodes|
+|nodeAttributes|Function[]|An array of functions that return VNodeProperties to be applied to the VNode|
+
+By default the base widget class applies `id`, `classes` and `styles` from the widgets specified `state` (either by direct state injection or via an observable store).
+
+#### Simple Widgets
+To create a basic widget `createWidgetBase` can be used directly by importing the class.
+
+```ts
+import createWidgetBase from 'dojo-widgets/bases/createWidgetBase';
+
+const myBasicWidget = createWidgetBase();
+```
+
+Creates the following DOM element:
```

Comment:
Maybe add an extra word or two, such as "This basic widget creates the follow DOM element:"